src/cluster/brokerPool.js

```diff
@@ -122,6 +125,11 @@ module.exports = class BrokerPool {
     const broker = await this.findConnectedBroker()
     const { host: seedHost, port: seedPort } = this.seedBroker.connection
 
+    // clear expired entries to prevent cache bloat
```

Comment:
@Nevon Regarding:
> I think your approach is valid. A simple topic -> metadata map with an expiry key should be enough, I think. We might have to think of an expiration mechanism though, to make sure that we eventually get rid of old metadata for topics that are no longer required.

I added this code as part of `refreshMetadata` for this reason. `refreshMetadata` already takes place on semi-regular intervals due to the `metadataMaxAge` setting. It's not a particularly tight mechanism to manage cache expiry, but it will prevent uncontrolled growth.